core/state_transition.go

```diff
@@ -254,10 +254,10 @@ func (st *StateTransition) buyGas(gasBailout bool) error {
 		if overflow {
 			return fmt.Errorf("%w: address %v", ErrInsufficientFunds, st.msg.From().Hex())
 		}
-		balanceCheck, overflow = balanceCheck.AddOverflow(balanceCheck, st.value)
-		if overflow {
-			return fmt.Errorf("%w: address %v", ErrInsufficientFunds, st.msg.From().Hex())
-		}
+	}
+	balanceCheck, overflow = balanceCheck.AddOverflow(balanceCheck, st.value)
```

Comment:
what does this do?
how does it fixes the issue, especially taking into account that we don't break on syncing this block

Reply:
The equation error message states that it should be `gas*gasPrice + value`, but how we had it before was that if the transaction is Legacy then it would be `gas*gasPrice` since it has no `gasFeeCap`

Reply:
I don't mind changing it back, so far it has not presented a problem.

Reply:
I would, just in case, sync eth mainnet on this branch and check if there are any inconsistencies.

Reply:
Doing that now. I synced Goerli, but mainnet would be more accurate. Either way I can revert it so we merge the PR since as of now Parlia trace doesn't work, so we can get that working. what do you think?